### PR TITLE
blas/gonum: add Ztpmv

### PR DESCRIPTION
**Pull request by vladimir-ch**

### CODE REVIEW

File: blas/gonum/level2cmplx128.go [hunk 1]
```diff
@@ -939,6 +939,227 @@ func (Implementation) Zhpr2(uplo blas.Uplo, n int, alpha complex128, x []complex
 	}
 }
 
+// Ztpmv performs one of the matrix-vector operations
+//  x := A * x    if trans = blas.NoTrans
```
Comment (kortschak) [inline blas/gonum/level2cmplx128.go]: I don't think we use the ':=' update operator in the godoc.

Comment (vladimir-ch) [reply blas/gonum/level2cmplx128.go]: It is being used at a couple of places. I'll send a PR to use `=` later.

File: blas/testblas/ztrmv.go [hunk 2]
```diff
@@ -11,221 +11,236 @@ import (
...
+	wantUnitConjTrans    []complex128
+	wantUnitConjTransNeg []complex128
+}{
+	{uplo: blas.Upper},
+	{uplo: blas.Lower},
+	{
+		uplo: blas.Upper,
+		a: []complex128{
+			6 - 8i, -10 + 10i, -6 - 3i, -1 - 8i,
+			0, 7 + 8i, -7 + 9i, 3 + 6i,
```
[the remark this answers is not in the document]
Comment (vladimir-ch) [reply blas/testblas/ztrmv.go]: I didn't want to add a package-level `znan` variable, and using `cmplx.NaN()` would obscure the contents of the matrix. Are you ok with having the `znan` variable?

[the remark this answers is not in the document]
Comment (kortschak) [reply blas/testblas/ztrmv.go]: `znan` sounds fine to me.

File: blas/gonum/level2cmplx128.go (hunk 1 above)
Comment (kortschak) [reply blas/gonum/level2cmplx128.go]: SGTM

File: blas/testblas/ztrmv.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (vladimir-ch) [reply blas/testblas/ztrmv.go]: I've added `znan` and use it in tests in place of `cmplx.NaN()`. PTAL

**Comment by kortschak** [comment]
Please remember to update netlib appropriately.

**Comment by vladimir-ch** [comment]
Oh, thanks for the reminder, I would surely forget otherwise.

**Pull request merged**